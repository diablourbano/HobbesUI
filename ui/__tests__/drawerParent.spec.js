import React from 'react';
import { shallow } from 'enzyme';
import DrawerSection from '../drawer/drawerParent';

describe('DrawerSection', () => {
  it('should render with default values', () => {
    const component = shallow((
      <DrawerSection>
        This is something
      </DrawerSection>
    ));

    expect(component).toMatchSnapshot();
  });

  it('should render it collapsed', () => {
    const component = shallow((
      <DrawerSection
        shouldCollapsed
        title="foo"
      >
        This is something
      </DrawerSection>
    ));

    expect(component).toMatchSnapshot();
  });
});
