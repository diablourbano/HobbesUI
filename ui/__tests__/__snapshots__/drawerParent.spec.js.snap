// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerSection should render it collapsed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrawerSection
    levelPosition={1}
    sectionColor="#8b939e"
    sectionFontSize={20}
    sectionHeight={32}
    shouldCollapsed={true}
    title="foo"
  >
    This is something
  </DrawerSection>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styles__ParentSection
          onPress={[Function]}
          sectionColor="#8b939e"
          sectionHeight={32}
        >
          <styles__ParentName
            sectionFontSize={20}
          >
            foo
          </styles__ParentName>
        </styles__ParentSection>,
        "This is something",
      ],
      "isCollapsed": true,
      "levelPosition": 1,
      "sectionColor": "#8b939e",
      "sectionHeight": 32,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styles__ParentName
            sectionFontSize={20}
          >
            foo
          </styles__ParentName>,
          "onPress": [Function],
          "sectionColor": "#8b939e",
          "sectionHeight": 32,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "sectionFontSize": 20,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        "type": [Function],
      },
      "This is something",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styles__ParentSection
            onPress={[Function]}
            sectionColor="#8b939e"
            sectionHeight={32}
          >
            <styles__ParentName
              sectionFontSize={20}
            >
              foo
            </styles__ParentName>
          </styles__ParentSection>,
          "This is something",
        ],
        "isCollapsed": true,
        "levelPosition": 1,
        "sectionColor": "#8b939e",
        "sectionHeight": 32,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styles__ParentName
              sectionFontSize={20}
            >
              foo
            </styles__ParentName>,
            "onPress": [Function],
            "sectionColor": "#8b939e",
            "sectionHeight": 32,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "foo",
              "sectionFontSize": 20,
            },
            "ref": null,
            "rendered": "foo",
            "type": [Function],
          },
          "type": [Function],
        },
        "This is something",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`DrawerSection should render with default values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrawerSection
    levelPosition={1}
    sectionColor="#8b939e"
    sectionFontSize={20}
    sectionHeight={32}
    title=""
  >
    This is something
  </DrawerSection>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styles__ParentSection
          onPress={[Function]}
          sectionColor="#8b939e"
          sectionHeight={32}
        >
          <styles__ParentName
            sectionFontSize={20}
          >
            
          </styles__ParentName>
        </styles__ParentSection>,
        "This is something",
      ],
      "isCollapsed": true,
      "levelPosition": 1,
      "sectionColor": "#8b939e",
      "sectionHeight": 32,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styles__ParentName
            sectionFontSize={20}
          >
            
          </styles__ParentName>,
          "onPress": [Function],
          "sectionColor": "#8b939e",
          "sectionHeight": 32,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "sectionFontSize": 20,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      "This is something",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styles__ParentSection
            onPress={[Function]}
            sectionColor="#8b939e"
            sectionHeight={32}
          >
            <styles__ParentName
              sectionFontSize={20}
            >
              
            </styles__ParentName>
          </styles__ParentSection>,
          "This is something",
        ],
        "isCollapsed": true,
        "levelPosition": 1,
        "sectionColor": "#8b939e",
        "sectionHeight": 32,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styles__ParentName
              sectionFontSize={20}
            >
              
            </styles__ParentName>,
            "onPress": [Function],
            "sectionColor": "#8b939e",
            "sectionHeight": 32,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "sectionFontSize": 20,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        "This is something",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
